src/Tickets/QR/Controller.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace TEC\Tickets\QR;
+
+use \TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+
+/**
+ * Class Controller.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\QR
+ */
+class Controller extends Controller_Contract {
+	/**
+	 * Determines if this controller will register.
+	 * This is present due to how UOPZ works, it will fail if method belongs to the parent/abstract class.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_active(): bool {
+		return true;
+	}
+
+	/**
+	 * Boot the Controller.
+	 *
+	 * This function is used to instantiate the singleton classes and register any other providers.
+	 *
+	 * @since   TBD
+	 */
+	public function boot() {
+		$this->container->bind( Facade::class, [ $this, 'bind_facade_or_error' ] );
+	}
+
+	/**
+	 * Register the controller.
+	 *
+	 * @since   TBD
+	 */
+	public function do_register(): void {
+		$this->boot();
```

Comment:
Avoid overriding the default `boot` method if there is nothing to really boot. What this controller should do, is just register a binding, see my previous comment.